Lib/test/test_pty.py

```diff
@@ -81,28 +101,74 @@ def setUp(self):
+    @expectedFailureIfStdinIsTTY
+    def test_openpty(self):
+        try:
+            mode = tty.tcgetattr(pty.STDIN_FILENO)
+        except tty.error:
+            # pty.STDIN_FILENO not a tty?
+            debug("tty.tcgetattr(pty.STDIN_FILENO) failed")
+            mode = None
+
+        try:
+            TIOCGWINSZ = tty.TIOCGWINSZ
+            TIOCSWINSZ = tty.TIOCSWINSZ
+        except AttributeError:
+            debug("TIOCSWINSZ/TIOCGWINSZ not available")
+            winsz = None
+        else:
+            try:
+                debug("Setting pty.STDIN_FILENO window size")
+                current_stdin_winsz = os.get_terminal_size(pty.STDIN_FILENO)
+
+                # Set number of columns and rows to be the
+                # floors of 1/5 of respective original values
+                winsz = struct.pack("HHHH", current_stdin_winsz.lines//5,
+                                    current_stdin_winsz.columns//5, 0, 0)
+                fcntl.ioctl(pty.STDIN_FILENO, TIOCSWINSZ, winsz)
+
+                # Were we able to set the window size
+                # of pty.STDIN_FILENO successfully?
+                s = struct.pack("HHHH", 0, 0, 0, 0)
+                new_stdin_winsz = fcntl.ioctl(pty.STDIN_FILENO, TIOCGWINSZ, s)
+                self.assertEqual(new_stdin_winsz, winsz,
+                                 "pty.STDIN_FILENO window size unchanged")
+            except OSError:
+                # pty.STDIN_FILENO not a tty?
+                debug("Failed to set pty.STDIN_FILENO window size")
+                winsz = None
+
         try:
-            debug("Calling master_open()")
-            master_fd, slave_name = pty.master_open()
-            debug("Got master_fd '%d', slave_name '%s'" %
-                  (master_fd, slave_name))
-            debug("Calling slave_open(%r)" % (slave_name,))
-            slave_fd = pty.slave_open(slave_name)
-            debug("Got slave_fd '%d'" % slave_fd)
+            debug("Calling pty.openpty()")
+            try:
+                master_fd, slave_fd = pty.openpty(mode, winsz)
+            except TypeError:
+                master_fd, slave_fd = pty.openpty()
+            debug("Got master_fd '%d', slave_fd '%d'" %
+                  (master_fd, slave_fd))
         except OSError:
             # " An optional feature could not be imported " ... ?
             raise unittest.SkipTest("Pseudo-terminals (seemingly) not functional.")
 
-        self.assertTrue(os.isatty(slave_fd), 'slave_fd is not a tty')
+        self.assertTrue(os.isatty(slave_fd), "slave_fd is not a tty")
+
+        if mode:
+            self.assertEqual(tty.tcgetattr(slave_fd), mode,
+                             "openpty() failed to set slave termios")
+        if winsz:
+            s = struct.pack("HHHH", 0, 0, 0, 0)
+            self.assertEqual(fcntl.ioctl(slave_fd, TIOCGWINSZ, s), winsz,
+                             "openpty() failed to set slave window size")
 
+        # RELEVANT ANYMORE?
```

Comment:
Maybe @ethanfurman or @Yhg1s could clarify, but I don't think we should add a new "RELEVANT ANYMORE?" comment.

Reply:
Thank you for noticing this. I agree that they are annoying; I apologize for this. Yes, I need clarification. The plan was to remove the annoying "RELEVANT ANYMORE?" comments after clarification.

Reply:
I myself actually did not put much thought into the blocks that I marked with "RELEVANT ANYMORE?"; I will think about them in greater detail on Saturday. They were just supposed to be there as reminders; anyway, they do not hurt the tests :)

Reply:
Aah. Now I realize why I added the extra "RELEVANT ANYMORE" here. Notice that the following comment mentions `pty.master_open()`, which opens the slave and subsequently closes it. As a result, the original `test_basic()` function was having to call `pty.slave_open()` to re-open the slave. However, the new `test_openpty()` calls `pty.openpty()`, which, unlike `pty.master_open()`, does not close the slave end. Anyway, that part does not seem to affect the test on Linux and FreeBSD.